src/components/input.js

```diff
@@ -80,6 +80,12 @@ export const TextInput = forwardRef(({ onChange, nativeOnChange = false, ...prop
   }, props))
 })
 
+export const LabeledTextInput = ({ onChange, ...props }) => {
```

Comment:
For this component I was thinking of something more along the lines of labeledCheckbox, where the developer must provide a label. The id would then be assigned to the label and aria-labeledby would be used to reference the label.

Looking at the existing uses of TextInput, I have some additional questions, and thinking through it some more, I feel the base TextInput component should be modified to identify whether the TextInput specified is accessible and render a label when appropriate. Looking at the existing assertion it seems that many of our text inputs are still not accessible, but I will have to check against the screen reader.

I think Brett had some thoughts on this, so it may be best for the three of us to get together and chat about it, maybe on Monday.